pkg/accounting/accounting.go

```diff
@@ -524,20 +548,52 @@ func (a *Accounting) getAccountingPeer(peer swarm.Address) *accountingPeer {
 	peerData, ok := a.accountingPeers[peer.String()]
 	if !ok {
 		peerData = &accountingPeer{
-			reservedBalance:       big.NewInt(0),
-			shadowReservedBalance: big.NewInt(0),
-			ghostBalance:          big.NewInt(0),
+			reservedBalance:         big.NewInt(0),
+			shadowReservedBalance:   big.NewInt(0),
+			ghostBalance:            big.NewInt(0),
+			totalDebtRepay:          big.NewInt(0),
+			paymentThreshold:        new(big.Int).Set(a.paymentThreshold),
+			paymentThresholdForPeer: new(big.Int).Set(a.paymentThreshold),
+			disconnectLimit:         new(big.Int).Set(a.disconnectLimit),
+			thresholdGrowAt:         new(big.Int).Set(a.thresholdGrowStep),
 			// initially assume the peer has the same threshold as us
-			paymentThreshold: new(big.Int).Set(a.paymentThreshold),
-			earlyPayment:     new(big.Int).Div(new(big.Int).Mul(a.paymentThreshold, big.NewInt(100-a.earlyPayment)), big.NewInt(100)),
-			connected:        false,
+			earlyPayment: new(big.Int).Div(new(big.Int).Mul(a.paymentThreshold, big.NewInt(100-a.earlyPayment)), big.NewInt(100)),
+			connected:    false,
 		}
 		a.accountingPeers[peer.String()] = peerData
 	}
 
 	return peerData
 }
 
+func (a *Accounting) NotifyPaymentThresholdUpgrade(peer swarm.Address, accountingPeer *accountingPeer) {
+
+	thresholdGrowChange := new(big.Int).Set(a.thresholdGrowChange)
+	if !accountingPeer.fullNode {
+		thresholdGrowChange.Set(a.lightThresholdGrowChange)
+	}
+
+	if accountingPeer.thresholdGrowAt.Cmp(thresholdGrowChange) >= 0 {
+		accountingPeer.thresholdGrowAt = new(big.Int).Mul(accountingPeer.thresholdGrowAt, big.NewInt(2))
+	} else {
+		if accountingPeer.fullNode {
+			accountingPeer.thresholdGrowAt = new(big.Int).Add(accountingPeer.thresholdGrowAt, a.thresholdGrowStep)
+		} else {
+			accountingPeer.thresholdGrowAt = new(big.Int).Add(accountingPeer.thresholdGrowAt, a.lightThresholdGrowStep)
+		}
+	}
+
+	refreshRate := new(big.Int).Set(a.refreshRate)
+	if !accountingPeer.fullNode {
+		refreshRate = new(big.Int).Set(a.lightRefreshRate)
+	}
+
+	accountingPeer.paymentThresholdForPeer = new(big.Int).Add(accountingPeer.paymentThresholdForPeer, refreshRate)
+	accountingPeer.disconnectLimit = new(big.Int).Div(new(big.Int).Mul(accountingPeer.paymentThresholdForPeer, big.NewInt(100+a.paymentTolerance)), big.NewInt(100))
+
+	_ = a.pricing.AnnouncePaymentThreshold(context.Background(), peer, accountingPeer.paymentThresholdForPeer)
```

Comment:
should we not at least log this? or is this already done elsewhere?

Reply:
not addressed yet

Reply:
it is

Reply:
you can only see it in the actual files changed tab, for some reason Conversation page displays a snippet of a different earlier version